crawl-build/update-public-repository.sh

```diff
@@ -20,7 +20,8 @@ update-crawl-ref() {
     say "Updating git repository $REPO_DIR"
     ( cd $REPO_DIR && git checkout -f &&
         git checkout $BRANCH &&
-        git pull )
+        git fetch &&
+        git reset --hard FETCH_HEAD )
```

Comment:
Is there a reason to use a hard reset over `git pull --rebase`? I *think* they should be the same for our normal use case, and also handle if someone has a local commit of some kind (which regular servers shouldn't do), though I haven't tested this.